src/cf-base.less

```diff
@@ -140,138 +140,132 @@ body {
     margin-bottom: unit( 19px / @font-size, em );
     font-size: unit( @font-size / @base-font-size-px, em );
     line-height: unit( @line-height / @font-size );
-
-    .respond-to-max(@mobile-max, {
-        .h3();
-    });
 }
 
-h3,
-.h3 {
-    @font-size: 22px;
-    @line-height: 28px;
-
-    margin-top: 0;
+.heading-3( @font-size: 22px, @line-height: 28px )  {
+    .webfont-regular();
     // Spacing in the design manual is based off of the baseline so this margin
     // number has been eyeballed to be 30px.
     margin-bottom: unit( 21px / @font-size, em );
     font-size: unit( @font-size / @base-font-size-px, em );
     line-height: unit( @line-height / @font-size );
-
-    .respond-to-max(@mobile-max, {
-        .h4();
-    });
 }
 
-h4,
-.h4 {
-    @font-size: 18px;
-
-    margin-top: 0;
+.heading-4( @font-size: 18px, @line-height: @base-line-height-px )  {
+    .webfont-medium();
     // Spacing in the design manual is based off of the baseline so this margin
     // number has been eyeballed to be 30px.
     margin-bottom: unit( 21px / @font-size, em );
     font-size: unit( @font-size / @base-font-size-px, em );
-    .webfont-medium();
-    line-height: unit( @base-line-height-px / @font-size );
+    line-height: unit( @line-height / @font-size );
 }
 
-h5,
-h6,
-.h5,
-.h6 {
+.heading-5( @font-size: 14px, @line-height: @base-line-height-px )  {
     .webfont-demi();
 
     letter-spacing: 1px;
     text-transform: uppercase;
-}
-
-h5,
-.h5 {
-    @font-size: 14px;
-
-    margin-top: 0;
     // Spacing in the design manual is based off of the baseline so this margin
     // number has been eyeballed to be 15px.
     margin-bottom: unit( 5px / @font-size, em );
     font-size: unit( @font-size / @base-font-size-px, em );
-    line-height: unit( @base-line-height-px / @font-size );
+    line-height: unit( @line-height / @font-size );
 }
 
-h6,
-.h6 {
-    @font-size: 12px;
+.heading-6( @font-size: 12px, @line-height: @base-line-height-px )  {
+    .webfont-demi();
 
-    margin-top: 0;
+    letter-spacing: 1px;
+    text-transform: uppercase;
     // Spacing in the design manual is based off of the baseline so this margin
     // number has been eyeballed to be 15px.
     margin-bottom: unit( 5px / @font-size, em );
     font-size: unit( @font-size / @base-font-size-px, em );
-    line-height: unit( @base-line-height-px / @font-size );
+    line-height: unit( @line-height / @font-size );
 }
 
-.subheader {
-    // For longer subheaders that we don't want to be too large or heavier than
-    // the regular weight.
-    @font-size: 18px;
-
+h1,
+h2,
+h3,
+h4,
+h5,
+h6 {
     margin-top: 0;
-    // Spacing in the design manual is based off of the baseline so this margin
-    // number has been eyeballed to be 30px.
-    margin-bottom: unit( 21px / @font-size, em );
-    font-size: unit( @font-size / @base-font-size-px, em );
+}
+
+h1,
+.h1 {
+    .heading-1();
+
+    .respond-to-max(@mobile-max, {
+        .heading-2();
+    });
```

Comment:
I tried to match what was already here, so I went no spaces on the mq mixins since the existing ones didn't use spaces. If we're going to change always spaces, we should make it a rule in the Front End playbook and update everything at once.